packages/vite/src/node/plugins/css.ts

```diff
@@ -1949,101 +2024,168 @@ function fixScssBugImportValue(
 }
 
 // .scss/.sass processor
-const scss: SassStylePreprocessor = async (
-  source,
-  root,
-  options,
-  resolvers,
+const makeScssWorker = (
+  resolvers: CSSAtImportResolvers,
+  alias: Alias[],
+  maxWorkers: number | undefined,
 ) => {
-  const render = loadPreprocessor(PreprocessLang.sass, root).render
-  // NOTE: `sass` always runs it's own importer first, and only falls back to
-  // the `importer` option when it can't resolve a path
-  const internalImporter: Sass.Importer = (url, importer, done) => {
+  const internalImporter = async (
+    url: string,
+    importer: string,
+    filename: string,
+  ) => {
     importer = cleanScssBugUrl(importer)
-    resolvers.sass(url, importer).then((resolved) => {
-      if (resolved) {
-        rebaseUrls(
+    const resolved = await resolvers.sass(url, importer)
+    if (resolved) {
+      try {
+        const data = await rebaseUrls(
           resolved,
-          options.filename,
-          options.alias,
+          filename,
+          alias,
           '$',
           resolvers.sass,
         )
-          .then((data) => done?.(fixScssBugImportValue(data)))
-          .catch((data) => done?.(data))
-      } else {
-        done?.(null)
+        return fixScssBugImportValue(data)
+      } catch (data) {
+        return data
       }
-    })
-  }
-  const importer = [internalImporter]
-  if (options.importer) {
-    Array.isArray(options.importer)
-      ? importer.unshift(...options.importer)
-      : importer.unshift(options.importer)
+    } else {
+      return null
+    }
   }
 
-  const { content: data, map: additionalMap } = await getSource(
-    source,
-    options.filename,
-    options.additionalData,
-    options.enableSourcemap,
-  )
-  const finalOptions: Sass.Options = {
-    ...options,
-    data,
-    file: options.filename,
-    outFile: options.filename,
-    importer,
-    ...(options.enableSourcemap
-      ? {
-          sourceMap: true,
-          omitSourceMapUrl: true,
-          sourceMapRoot: path.dirname(options.filename),
+  const worker = new WorkerWithFallback(
+    () =>
+      async (
+        sassPath: string,
+        data: string,
+        // additionalData can a function that is not cloneable but it won't be used
+        options: SassStylePreprocessorOptions & { additionalData: undefined },
+      ) => {
+        // eslint-disable-next-line no-restricted-globals
+        const sass: typeof Sass = require(sassPath)
+        // eslint-disable-next-line no-restricted-globals
+        const path = require('node:path')
```

Comment:
It would be nice to comment how `require` works here (and in the other workers).